src/Report/EnabledMembers.php

```diff
@@ -0,0 +1,206 @@
+        $params = array_filter($params, 'strlen');
+
+        // Having a WHERE limits the results, which means the the COUNT will be off (GROUP BY applied after WHERE)
+        // So clip the count filter and apply it after the database query, only if Methods filter is also set
+        // this way we can delegate to the database where possible, which should be faster.
+        if (isset($params['Count'])) {
+            $desiredCount = (int)$params['Count'];
+            if (isset($params['Methods'])) {
+                $countFilter = $desiredCount;
+                unset($params['Count']);
+            } elseif ($desiredCount) {
+                // When querying SQL COUNT() - this will always factor in the backup method, which is excluded in the
+                // secondary query below to get the list of method names the user has registered. So to compensate we
+                // need to bump up the entered number by 1.
+                $params['Count'] = $desiredCount + 1;
+            }
+        }
+
+        $filteredList = ArrayList::create([]);
+        $backupClass = get_class(singleton(MethodRegistry::class)->getBackupMethod());
+        /** @var Member[]&MemberExtension[] $members */
+        $members = $this->applyParams(Member::get(), $params)->toArray();
+        foreach ($members as $member) {
+            $defaultMethod = $member->getDefaultRegisteredMethod();
+            $defaultMethodClassName = $defaultMethod ? $defaultMethod->MethodClassName : '';
+
+            $registeredMethods = $member
+                ->RegisteredMFAMethods()
+                ->exclude('MethodClassName', $backupClass)
+                ->column('MethodClassName');
+            $registeredMethodNames = array_map(function (string $methodClass) use ($methods) {
+                return $methods[$methodClass];
+            }, $registeredMethods);
+
+            $memberReportData = ArrayData::create();
+            foreach ($member->summaryFields() as $field => $name) {
+                $memberReportData->$field = $member->$field;
+            }
+
+            $memberReportData->methodCount = (string)(count($registeredMethods) ?: '');
+            $memberReportData->registeredMethods = implode(', ', $registeredMethodNames);
+            $memberReportData->defaultMethodName = $methods[$defaultMethodClassName] ?? '';
+            $memberReportData->skippedRegistration = $member->HasSkippedMFARegistration
+                ? $member->dbObject('HasSkippedMFARegistration')->Nice()
+                : '';
+
+            $filteredList->push($memberReportData);
+        };
+        if (isset($countFilter)) {
+            $filteredList = $filteredList->filter('methodCount', $countFilter);
+        }
+        $this->extend('updateSourceRecords', $filteredList);
+        return $filteredList;
+    }
+
+    public function columns()
+    {
+        $columns = singleton(Member::class)->summaryFields() + [
```

Comment:
Won't this mean that any customisations to Member GridField summary fields will change the output of this report as well? I think you should make it explicit instead - i.e. ID, first name, surname, email

Reply:
I disagree. I have formatted it this way intentionally to allow alterations to Member summary fields to be reflected in this report. E.g. if someone were to configure the Member object to use a different unique identifier than the Email field, etc.

Reply:
Ok, fair enough =)